src/modules/twap/utils/updateTwapOrdersList.ts

```diff
@@ -0,0 +1,24 @@
+import { TWAP_PENDING_STATUSES } from '../const'
+import { TwapOrdersList } from '../state/twapOrdersListAtom'
+
+/**
+ *
+ * This function merge current state of TWAP orders list with a new one
+ * It only updates an order if it in a pending state
+ */
+export function updateTwapOrdersList(currentState: TwapOrdersList, nextState: TwapOrdersList): TwapOrdersList {
+  // Filter nextState and left only orders with pending statuses
+  const newState = Object.keys(nextState).reduce<TwapOrdersList>((acc, orderId) => {
+    const currentOrder = currentState[orderId]
+
+    // Update an order only if it's in pending state
+    // Then don't update it
+    if (!currentOrder || TWAP_PENDING_STATUSES.includes(currentOrder.status)) {
```

Comment:
Shouldn't you do this if `currentOrder` exists?

The comment only mentions if it's in a pending state, but there's no mention about the currentState order

```suggestion
    if (currentOrder || TWAP_PENDING_STATUSES.includes(currentOrder.status)) {
```

Reply:
Ahhh ok, now I got it!